### PR TITLE
Add configurable privacy statement link

### PR DESCRIPTION
**Pull request by lampajr**
## Fixes Issue

Fixes https://github.com/Hyperfoil/Horreum/issues/1408

## Changes proposed

Provide additional `privacy` statement in the `About` popup:

image

The link/URL is configurable at installation/build level using property `horreum.privacy`.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by lampajr** [comment]
There are still a couple of open points:
- [ ] Set a reasonable default value, right now I am using `/link/to/privacy/statement`
- [ ] Is `Privacy` category name fine?

I kept the new `VersionInfo.privacyStatement` field optional from OpenAPI perspective in order to be backward compatible, the webapp should manage when the field is `undefined`.

File: horreum-web/package-lock.json
Comment (johnaohara) [inline horreum-web/package-lock.json:1]: Can you rebase on top of current main branch? the `package-lock.json` issue should be resolved now

File: horreum-web/src/About.tsx [hunk 1]
```diff
@@ -94,6 +98,9 @@ export default function About() {
                         rows={[
                             ["Version", versionInfo.version],
                             ["Up since", formatDateTime(versionInfo.startTimestamp)],
+                            versionInfo.privacyStatement
+                                ? ["Privacy",  <TableText><a href={versionInfo.privacyStatement}>{versionInfo.privacyStatement}</a></TableText>]
```
Comment (johnaohara) [inline horreum-web/src/About.tsx:102]:
```suggestion
                                ? ["Privacy Statement",  <TableText><a href={versionInfo.privacyStatement}>{versionInfo.privacyStatement}</a></TableText>]
```

**Review by johnaohara — CHANGES_REQUESTED** [review]
Please can you rebase

**Comment by johnaohara** [comment]
> There are still a couple of open points:
> 
>     * [ ]  Set a reasonable default value, right now I am using `/link/to/privacy/statement`. Any suggestion?
> 

I am not sure if there is a reasonable default, it is probably best to leave this either commented out in application.properties and not display anything by default. Users can see that it is an option that can be set and/or we need an admin guide that covers the setting

>     * [ ]  Is `Privacy` category name fine on UI?
> 
> 
> I kept the new `VersionInfo.privacyStatement` field optional from OpenAPI perspective in order to be backward compatible, the webapp should manage when the field is `undefined`.

I think it is fine to be a configuration option for Horreum as it should apply system wide to all users of the installation

File: horreum-web/package-lock.json
Comment (lampajr) [reply horreum-web/package-lock.json:1]: It should be already rebased on top of `master` :thinking:

**Review by johnaohara — APPROVED** [review]

File: horreum-web/package-lock.json
Comment (johnaohara) [reply horreum-web/package-lock.json:1]: There was a commit that fixed `package-lock.json`. Looks now

Comment (lampajr) [reply horreum-web/package-lock.json:1]: Oh ok, I missed that :facepalm: 
Anyway I rebased again just to be sure :heavy_check_mark:

**Comment by johnaohara** [comment]
merged, thank you for your contribution!

**Comment by lampajr** [comment]
@johnaohara do we want to address https://github.com/Hyperfoil/Horreum/pull/1437#issuecomment-1983178553 in a separate PR? because otherwise by default the webpage will show `/link/to/privacy/statement` as privacy statement which does not make sense

**Comment by johnaohara** [comment]
> @johnaohara do we want to address [#1437 (comment)](https://github.com/Hyperfoil/Horreum/pull/1437#issuecomment-1983178553) in a separate PR? because otherwise by default the webpage will show `/link/to/privacy/statement` as privacy statement which does not make sense

yes please, i have already merged this PR

**Pull request merged**